src/pages/Folder.jsx

```diff
@@ -0,0 +1,79 @@
+import { getData, getFolderListData, getWholeFolderListData } from "../api/api";
+import { useEffect, useState } from "react";
+import CardBox from "../components/CardBox";
+import TopNavBar from "../components/TopNavBar";
+import SearchBar from "../components/SearchBar";
+import BottomNavBar from "../components/BottomNavBar";
+import AddLinkBar from "../components/AddLinkBar";
+import FolderListBar from "../components/FolderListBar";
+import EmptyFolder from "../components/EmptyFolder";
+
+function Folder() {
+  const [profileData, setProfileData] = useState({});
+  const [folderList, setFolderList] = useState([]);
+  const [linksData, setLinksData] = useState([]);
+
+  const getProfileData = async (options) => {
+    try {
+      const newProfile = await getData(options);
+      setProfileData(newProfile);
+    } catch (err) {
+      console.error(err);
+      setProfileData({});
+    }
+  };
+
+  const getFolderList = async () => {
+    try {
+      const folderListData = await getFolderListData();
+      const { data } = folderListData;
+      setFolderList(data);
+    } catch (err) {
+      console.error(err);
+      setFolderList([]);
+    }
+  };
+
+  const getLinks = async (folderId) => {
+    try {
+      console.log("folderId: ", folderId);
+      const linksData = await getWholeFolderListData(folderId);
+      const { data } = linksData;
+      setLinksData(data);
+    } catch (err) {
+      console.error(err);
+      setLinksData([]);
+    }
+  };
+
+  useEffect(() => {
+    getProfileData({ path: "user" });
+    getFolderList();
+    getLinks({ folderId: "1" });
+  }, []);
+
+  return (
+    <>
+      <header>
+        <TopNavBar profileData={profileData} />
+        <AddLinkBar />
+      </header>
+      <main>
+        <section>
+          <SearchBar />
+          <FolderListBar folderList={folderList} onClick={getLinks} />
+          {linksData.length === 0 ? (
```

Comment:
linksData 가 응답이 제대로 안 올경우 undefined 가 될 수 있는데,
이 경우 undefined.length 라서 아예 에러가 날 수 있습니다!
그래서 undefined 인 경우 length 가 0 일때와 동일한 처리는 해주는 식의 코드를 추천드려요
```suggestion
          {linksData?.length ? (
```